docs/task_manager.rst

```diff
@@ -0,0 +1,97 @@
+Background Task Processing
+######################################
+
+.. currentmodule:: task_manager
+
+Introduction
+======================================
+
+In order to build responsive Toolkit apps, work needs to be scheduled to run
+in background threads. The :class:`BackgroundTaskManager` handles generic
+background processing and is used by many of the Toolkit frameworks.
+
+When for example using a :class:`~shotgun_model.ShotgunModel`, the model
```

Comment:
Wording suggestion: "When using a :class:`~shotgun_model.ShotgunModel`, for example, ..."